src/components/Auth/Login.tsx

```diff
@@ -93,6 +99,21 @@ const Login = (props: LoginProps) => {
   const [otp, setOtp] = useState("");
   const [otpError, setOtpError] = useState<string>("");
   const [otpValidationError, setOtpValidationError] = useState<string>("");
+  const [resendOtpCountdown, setResendOtpCountdown] =
+    useState(resendOtpTimeout);
+
+  // Timer Function for resend OTP
+  useEffect(() => {
+    if (resendOtpCountdown <= 0) {
+      return;
+    }
+
+    const timer = setInterval(() => {
+      setResendOtpCountdown((prevTime) => prevTime - 1);
+    }, 1000);
+
+    return () => clearInterval(timer);
+  }, [resendOtpCountdown]);
```

Comment:
we don't need to add resendOtpCountdown as a dependency here; that would result in creating new intervals every second.